Requires typescript in non typescript project

Running this plugin on a non typescript problem yielded the following error

```
Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/node_modules/typescript-eslint-parser/parser.js:11:10)
```

Installing typescript fixes the problem but it should not be required on a non typescript project.

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
   "dependencies": {
     "babel-traverse": "^6.16.0",
     "babylon": "^6.12.0",
-    "typescript": "^2.3.4",
+    "typescript": "~2.3.2",
     "typescript-eslint-parser": "^3.0.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -34,12 +34,12 @@
     "eslint-plugin-import": "^2.0.1",
     "jest": "^16.0.2",
     "lodash": "^4.15.0",
-    "stylelint": "^7.3.0",
-    "typescript": "^2.3.4"
+    "stylelint": "^7.3.0"
   },
   "dependencies": {
     "babel-traverse": "^6.16.0",
     "babylon": "^6.12.0",
+    "typescript": "^2.3.4",
     "typescript-eslint-parser": "^3.0.0"
   }
 }

